internal/ethereum/blocklistener.go

```diff
@@ -92,6 +100,34 @@ func newBlockListener(ctx context.Context, c *ethConnector, conf config.Section,
 	return bl, nil
 }
 
+// setting block filter status updates that new block filter has been created
+func (bl *blockListener) setBlockFilterStatus() {
+	bl.blockFilterEstablished.mux.Lock()
+	defer bl.blockFilterEstablished.mux.Unlock()
+	if !bl.blockFilterEstablished.isEstablished {
+		bl.blockFilterEstablished.isEstablished = true
+		close(bl.blockFilterEstablished.signal)
+	}
+}
+
+// un-setting block filter status updates that no block filter is present on block listener
+func (bl *blockListener) unsetBlockFilterStatus() {
+	bl.blockFilterEstablished.mux.Lock()
+	defer bl.blockFilterEstablished.mux.Unlock()
+	if bl.blockFilterEstablished.isEstablished {
+		bl.blockFilterEstablished.isEstablished = false
+		bl.blockFilterEstablished.signal = make(chan struct{})
+	}
+}
+
+func (bl *blockListener) blockTillBlockFilterEstablished(ctx context.Context) {
```

Comment:
I suggest removing `ctx context.Context` from this function as it's not used in the way you described. @rodion-lim-partior To enable it to work in the way you described, you'll need to surface this extra parameter to the top level. For now, in the code, you just used `context.Background()` which only gets canceled when the program exit, seems a bit redundant to `bl.ctx.Done()`.